### DISCUSSION TITLE
Ghidra consistently showing "0x0" instead of the register name

### DISCUSSION THREAD

**Discussion by rpitasky** [NationalSecurityAgency/ghidra#6490] [Q&A]
I've tried to encapsulate the entire issue in this one screenshot; I'd like ghidra to say `PUSH IX` here or fill in the value of IX if IX is a known value at that point. It does this "incorrect" behavior very consistently, for tons of different instructions.
I assume I'm doing some fundamental thing wrong here! How can I fix this?

![image](https://github.com/NationalSecurityAgency/ghidra/assets/111201305/d5c4719b-0000-4115-93a0-f3f4ed22290b)

**Comment by GhidorahRex** [comment]
What is `IX16` here? Can you share all relevant parts of the slaspec? Registers, constructor, subconstructor?

**Reply by rpitasky** [reply]
I hate to dump the whole slaspec on you but I am willing to do so. Here's the relevant register definitions:

```
define register offset=0x30 size=1 [IXL IXH IXU IYL IYH IYU _ _ _ _ _ _ _ _ _ _ MBASE R];
define register offset=0x30 size=3 [IX24 IY24 SPL PC];
define register offset=0x30 size=2 IX16;
define register offset=0x33 size=2 IY16;
define register offset=0x3a size=2 SPS;
define register offset=0x3c size=2 I;
```

This architecture solves the very specific problem of legacy Z80 codebases not meeting demands for memory and speed. There are processor flag bits ("instruction long" `il` and "long" `l`) which can be independently set (either through an instruction opcode prefix as below or for an entire section of code through a prefixed jump) to force the usual 16 bit registers to be 24 bits.

Note that instruction prefixes are indicated with suffixes in the disassembly i.e. `LD.SIL`
```
:^instruction is phase=0 & op0_8=0x40; instruction [phase=1; suffixed=1; mode_l=0; mode_il=0;] {}
:^instruction is phase=0 & op0_8=0x49; instruction [phase=1; suffixed=1; mode_l=1; mode_il=0;] {}
:^instruction is phase=0 & op0_8=0x52; instruction [phase=1; suffixed=1; mode_l=0; mode_il=1;] {}
:^instruction is phase=0 & op0_8=0x5B; instruction [phase=1; suffixed=1; mode_l=1; mode_il=1;] {}
:^instruction is phase=0 & instruction [ phase=1; suffixed=0; ] {}

suffix: ".SIS" is suffixed=1 & mode_l=0 & mode_il=0 {}
suffix: ".LIS" is suffixed=1 & mode_l=1 & mode_il=0 {}
suffix: ".SIL" is suffixed=1 & mode_l=0 & mode_il=1 {}
suffix: ".LIL" is suffixed=1 & mode_l=1 & mode_il=1 {}
suffix: "" is suffixed=0 {}
```

If there's anything else I can provide that would be of use, please let me know.
